go-runtime/ftl/option.go

```diff
@@ -5,95 +5,90 @@ import (
 	"database/sql"
 	"database/sql/driver"
 	"encoding"
-	"encoding/json"
 	"fmt"
 	"reflect"
-
-	ftlencoding "github.com/TBD54566975/ftl/go-runtime/encoding"
 )
 
 // Stdlib interfaces types implement.
 type stdlib interface {
 	fmt.Stringer
 	fmt.GoStringer
-	json.Marshaler
-	json.Unmarshaler
 }
 
 // An Option type is a type that can contain a value or nothing.
 type Option[T any] struct {
-	value T
-	ok    bool
+	Val  T
```

Comment:
We _really_ shouldn't be doing this, it violates encapsulation completely and it's a public type exposed to end users.

Reply:
I added a proposal for fixing this here: https://github.com/TBD54566975/ftl/pull/1417#discussion_r1591761950

What do you think? Pending agreement, I'll fix forward using that strategy.